### ISSUE TITLE
question about scope of C and B

### ISSUE THREAD

**Issue by SugarMasuo** [cvlab-stonybrook/doc3D-dataset#2]
- Acoording to paper, predicted 3D coordibate ranges in [0,1]，but in demo.py, the range of three coordinate axes groundtruth are ：
xmx, xmn, ymx, ymn,zmx, zmn= 1.2539363, -1.2442188, 1.2396319, -1.2289206, 0.6436657, -0.67492497     ## value obtained from the entire dataset 
So, the groundtruth of 3D coordinate also should  be normalized to [0,1] while training, do I understand correctly?

- Acoording to paper and demo.py, the value of backward map should be the 2D coordinate of input image, however, predicted B ranges [-1,1] ? puzzled.

Looking forward your reply. Thank you !

**Comment by sagniklp** [comment]
- Yes, you are correct! You have to use range for [0-1] scaling.
- The backward map is 2D coordinates of input image, but it's scaled to [-1,1] for training purposes. `torch.nn.functional.grid_sample()` requires the map to be in [-1,1]. Follow line 115-116 for this scaling in `demo.py` - https://github.com/cvlab-stonybrook/doc3D-dataset/blob/f62c8bf14f5eb507db7f2d96ffc3474c544be7c7/demo.py#L115

**Comment by SugarMasuo** [comment]
I understand of range of two variables！ I have another question, how to get the backward map? I guess, it is calculated by stretch each grid cell in the warped mesh to plane with bilinear interpolation of course,

It's so nice to talk to you. Thank you  so much! Look for your reply.

**Comment by SugarMasuo** [comment]
I understand of range of two variables！ I have another question, how to get the backward map? I guess, it is calculated by stretch each grid cell in the warped mesh to plane with bilinear interpolation of course,

It's so nice to talk to you. Thank you  so much! Look for your reply.

**Comment by sagniklp** [comment]
Backward mapping can be calculated from the UV aka forward mapping. We use MatLab to calculate it. Following is an equivalent python function-
```python
import scipy.interpolate as spin
import numpy as np
def fm2bm(fm,s):
    '''
    fm: the forward mapping in range [0,1]
    s: shape of the image required to sample from, defines the range of image coordinates in 
    '''
    fm = fm * s;
    msk = fm[:, :, 0] > 0
    s2d= fm [msk]
    tx,ty = np.nonzero(msk)
    grid =np.meshgrid(np.linspace(1,s,s),np.linspace(1,s,s))
    vx=spin.griddata(s2d,tx, tuple(grid), method='nearest')/float(s)
    vy=spin.griddata(s2d,ty, tuple(grid), method='nearest')/float(s)
    bm=np.stack([vy,vx],axis=-1)
    return bm
```
